security/pkg/nodeagent/caClient/client.go

```diff
@@ -77,3 +101,50 @@ func (cl *caClient) CSRSign(ctx context.Context, csrPEM []byte, subjectID string
 
 	return ret, nil
 }
+
+// TODO(quanlin): remove below workaround function once CA put SubjectId field as optional.
+// token format - "Bearer ya29.****"
+type content struct {
+	ID string `json:"azp"`
+}
+
+const tokenPrefix = "Bearer "
+
+// Parse subjectID from token by sending web request and parsing the response content.
+// This function can be removed once CA put SubjectId field as optional.
+func parseSubjectID(token string) (string, error) {
+	if !strings.HasPrefix(token, tokenPrefix) {
+		log.Errorf("invalid token %q, should start with %q", token, tokenPrefix)
+		return "", errors.New("invalid token")
+	}
+
+	strs := strings.Fields(token)
+	url := fmt.Sprintf("https://www.googleapis.com/oauth2/v3/tokeninfo?access_token=%s", strs[1])
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
I thought I had added a comment about this line, but now can't find it....
Hard coding `InsecureSkipVerify` to `true` seems odd to me since it's not a secure default. Why do we need this to be `true`? Should it be a flag value instead of hardcoded?

Reply:
thanks, I agree in prod it shouldn't skip, just put as it for now to unblock integration; the remove parsesubjectID workaround in nodeagent will be removed very soon since CA team's change is under review.